kable-btleplug-ffi/src/characteristic.rs

```diff
@@ -0,0 +1,50 @@
+use crate::descriptor::Descriptor;
+use crate::uuid::Uuid;
+use btleplug::api::CharPropFlags;
+
+#[derive(Clone, uniffi::Record)]
+pub struct Characteristic {
+    pub uuid: Uuid,
+    pub service: Uuid,
+    pub properties: CharacteristicPropertyFlags,
+    pub descriptors: Vec<Descriptor>,
+}
+
+#[derive(Clone, uniffi::Record)]
+pub struct CharacteristicPropertyFlags {
+    bits: u8,
+}
+
+impl From<Characteristic> for btleplug::api::Characteristic {
+    fn from(value: Characteristic) -> Self {
+        btleplug::api::Characteristic {
+            uuid: value.uuid.into(),
+            service_uuid: value.service.into(),
+            properties: value.properties.into(),
+            descriptors: value.descriptors.into_iter().map(|x| x.into()).collect(),
+        }
+    }
+}
+
+impl From<btleplug::api::Characteristic> for Characteristic {
+    fn from(value: btleplug::api::Characteristic) -> Self {
+        Characteristic {
+            uuid: value.uuid.into(),
+            service: value.service_uuid.into(),
+            properties: value.properties.into(),
+            descriptors: value.descriptors.into_iter().map(|x| x.into()).collect(),
```

Comment:
nit:
```suggestion
            descriptors: value.descriptors.into_iter().map(btleplug::api::Descriptor::into).collect(),
```

Plenty more of these in other files but is a nit so I'll not comment on them all, sometimes it reads better, sometimes it doesn't and I don't have a strong opinion on unecessary lamda usage so use whatever style you prefer.